lib/index.js

```diff
@@ -193,11 +200,11 @@ Socket.prototype._startPollSend = function () {
 }
 
 Socket.prototype._startPollReceive = function () {
-    if (!this._pollReceive) {
-        this._pollReceive = nn.PollReceiveSocket(this.binding, function (events) {
-            if (events) this._receive();
-        }.bind(this));
-    }
+  if (!this._pollReceive) {
+    this._pollReceive = nn.PollReceiveSocket(this.binding, function () {
+      this.push(this._receive());
+    }.bind(this));
```

Comment:
Something doesn't seem right here.  Is it because we no longer inherit from EventEmitter that we no longer get the param events?  We should make sure nn.PollReceiveSocket doesn't ever invoke the callback with an argument that we should be checking for truthiness, like we were previously.